docs/json-map.md

```diff
@@ -0,0 +1,72 @@
+# Overview
+JSON Map is a JSON-based format which describes part of the structure of another JSON document. Maps inform clients of the sizes of parts of the JSON and to allow constant time construction of subsets of the document.
+
+For example, suppose you have a 20GB JSON file which contains a huge array with one million elements along with some surrounding document structure. You can construct a map with a 0.1% target size (20 MB). The map will identify the path to the huge array, the number of elements it has, where each nth array element begins, and where the array itself starts and ends. 
```

Comment:
>huge [](start = 64, length = 4)

Copyediting nit: "huge" isn't necessary. "One million elements" conveys the hugeness :-)

Reply:
fixed